### PR TITLE
Final Submission

### PR DESCRIPTION
**Pull request by AyushSinghal9020**
@chirag9127 @encrypted-soul

### CODE REVIEW

File: dec_2023/round_2/r2/Ayush_Singhal/Ayush_predictions.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    checkpoint = ModelCheckpoint('best_model.h5', save_best_only=True)
+    early_stopping = EarlyStopping(monitor='val_loss', patience=10)
+    history = model.fit(X_train, y_train, epochs=5, batch_size=32, validation_data=(X_val, y_val),
+                        callbacks=[checkpoint, early_stopping])
+    test_loss = model.evaluate(X_test, y_test)
+    y_pred = model.predict(X_test)
+    y_pred_original = scaler.inverse_transform(np.hstack((X_test[:, -1, :-1], y_pred.reshape(-1, 1))))
+    y_test_original = scaler.inverse_transform(np.hstack((X_test[:, -1, :-1], y_test.reshape(-1, 1))))
+    threshold = 0.02
+    cosine_sim = cosine_similarity(y_test.reshape(1, -1), y_pred.reshape(1, -1))[0, 0]
```
Comment (chirag9127) [inline dec_2023/round_2/r2/Ayush_Singhal/Ayush_predictions.py:101]: You need to convert your preds to a list of `ETHPriceRanges` values.

**Comment by chirag9127** [comment]
@AyushSinghal9020 can you add requirements.txt?

**Comment by AyushSinghal9020** [comment]
Dear Chirag,

Due to unexpected issues, we've trained a new model, causing a 9-hour delay past the hackathon deadline. Will you please consider accepting the updated model. We apologize for any inconvenience.

I have also added the requirments.txt file

**Comment by chirag9127** [comment]
@AyushSinghal9020 can you share your final output? It looks like your model will take a fair bit of time to run

**Pull request merged**